core/src/main/scala/org/allenai/common/Logging.scala

```diff
@@ -46,5 +52,107 @@ trait Logging {
       if (internalLogger.isErrorEnabled) {
         internalLogger.error(message, throwable)
       }
+
+    /** Simple logback configuration.
+      * Hopefully this will be discoverable by just typing <code>logger.config().[TAB]</code>
+      *
+      * Examples:
+      * <code>
+      * logger.config("org.apache.spark").setLevel(Level.WARN)
+      *
+      * logger.config().addAppender(
+      *   logger.factory.patternLayoutEncoder("%-5level [%thread]: %message%n"),
+      *   logger.factory.consoleAppender
+      * )
+      * </code>
+      *
+      * @param loggerName the logger name, by default ROOT.
+      */
+    def config(loggerName: String = org.slf4j.Logger.ROOT_LOGGER_NAME) = {
+      new Config(loggerName)
+    }
+
+    class Config(loggerName: String = org.slf4j.Logger.ROOT_LOGGER_NAME) {
+      private val logger: Logger = LoggerFactory.getLogger(loggerName).asInstanceOf[Logger]
+
+      /** Resets the logger. */
+      def reset() = {
+        logger.getLoggerContext.reset()
+      }
+
+      /** Simple log level setting. Example:
+        * <code>
+        * logger.config("org.apache.spark").setLevel(Level.WARN)
+        * </code>
+        */
+      def setLevel(level: Level) = {
+        logger.setLevel(level)
+      }
+
+      /** Simple log appender creation. Example:
+        * <code>
+        * logger.config().addAppender(
+        *   logger.factory.patternLayoutEncoder("%-5level [%thread]: %message%n"),
+        *   logger.factory.consoleAppender
+        * )
+        * logger.config().addAppender(
+        *   logger.factory.htmlLayoutEncoder("%relative%thread%level%logger%msg"),
+        *   logger.factory.fileAppender("./log.html")
+        * )
+        * </code>
+        */
+      def addAppender(
+        encoder: Encoder[ILoggingEvent],
+        appender: OutputStreamAppender[ILoggingEvent]
+      ) = {
+        val loggerContext = logger.getLoggerContext
+        encoder.setContext(loggerContext)
+        encoder.start()
+        appender.setContext(loggerContext)
+        appender.setEncoder(encoder)
+        appender.start()
+        logger.addAppender(appender)
+      }
+    }
+
+    /** Factory methods for some simple config objects. */
+    object factory {
```

Comment:
Done. 
"factory" was feeling odd for me too. I was missing that I can use ```case class Config``` instead of ```def config()``` + ```class Config```.